core/config/backend_config.go

```diff
@@ -548,5 +552,18 @@ func (c *BackendConfig) GuessUsecases(u BackendConfigUsecases) bool {
 		}
 	}
 
+	if (u & FLAG_TOKENIZE) == FLAG_TOKENIZE {
+		tokenizeCapableBackends := []string{"llama.cpp", "rwkv"}
+		if !slices.Contains(tokenizeCapableBackends, c.Backend) {
+			return false
+		}
+	}
+
+	if (u & FLAG_VAD) == FLAG_VAD {
+		if c.Backend != "silero-vad" {
```

Comment:
note for later: at some point we should think about having these constants defined. There are some repetitions as well in the model initializer

Reply:
Agreed - probably worth doing that as followup if you're OK with that

Reply:
definitely, was just a reminder/note